relayer/src/worker/client.rs

```diff
@@ -52,40 +58,47 @@ pub fn detect_misbehavior_task<ChainA: ChainHandle, ChainB: ChainHandle>(
 ) -> Option<TaskHandle> {
     if client.is_expired_or_frozen() {
         warn!(
-            "skipping detect misbehavior task on frozen client: {}",
-            client.id()
+            client = %client.id(),
+            "skipping detect misbehavior task on frozen client",
         );
         return None;
     }
 
     {
-        debug!("[{}] doing first misbehavior check", client);
+        let _span = span!(
+            tracing::Level::DEBUG,
+            "DetectMisbehaviorFirstCheck",
+            client = %client.id,
+            src_chain = %client.src_chain.id(),
+            dst_chain = %client.dst_chain.id(),
+        )
+        .entered();
+        debug!("doing first check");
         let misbehavior_result = client.detect_misbehaviour_and_submit_evidence(None);
-        debug!(
-            "[{}] detect misbehavior result: {:?}",
-            client, misbehavior_result
-        );
+        trace!("detect misbehavior result: {:?}", misbehavior_result);
     }
 
     let handle = spawn_background_task(
-        format!("DetectMisbehaviorWorker({})", client),
+        span!(
+            tracing::Level::ERROR,
+            "DetectMisbehaviorWorker",
+            client = %client.id,
+            src_chain = %client.src_chain.id(),
+            dst_chain = %client.dst_chain.id(),
+        ),
```

Comment:
This is a bit more repetitive than the previous approach, but this allows filtering logs by any of the individual ID fields.

If this level of detail is not desirable, we could implement `tracing::Value` on the objects that were logged with `Display` (removing the issue of implementing a general-purpose formatting trait just for logging purposes), and resurrect the previous compact string representation.